packages/fuel-indexer-api-server/src/uses.rs

```diff
@@ -293,6 +293,78 @@ pub(crate) async fn register_index_assets(
     Err(ApiError::default())
 }
 
+pub(crate) async fn get_nonce(
+    Extension(pool): Extension<IndexerConnectionPool>,
+) -> ApiResult<axum::Json<Value>> {
+    let mut conn = pool.acquire().await?;
+    let nonce = queries::create_nonce(&mut conn).await?;
+
+    Ok(Json(json!(nonce)))
+}
+
+pub(crate) async fn verify_signature(
+    Extension(config): Extension<IndexerConfig>,
+    Extension(pool): Extension<IndexerConnectionPool>,
+    Json(payload): Json<VerifySignatureRequest>,
+) -> ApiResult<axum::Json<Value>> {
+    if config.authentication.enabled {
+        let mut conn = pool.acquire().await?;
+        match config.authentication.strategy {
+            Some(AuthenticationStrategy::JWT) => {
+                let nonce = queries::get_nonce(&mut conn, &payload.message).await?;
+                queries::delete_nonce(&mut conn, &nonce).await?;
+
+                let mut buff: [u8; 64] = [0u8; 64];
+                buff.copy_from_slice(&payload.signature.as_bytes()[..64]);
```

Comment:
This ends up copying the whole signature anyways; see [this example](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=336b9b9274fb64f1759fd889651250c7). You could get away with just doing `buff.copy_from_slice(&payload.signature.as_bytes()[..])` unless we plan on adding something to the end of the signature in the near future.

Reply:
@deekerno 
- So I tried your suggestion and turns out the signature (or at least the digest/string that is passed in the auth request)  is larger than the 64 bytes required to build a `Signature`
- @kayagokalp 
  - Am I resolving the `Signature` here properly using the output from `forc-wallet`? That `payload.signature` is the signature output from `forc-wallet` signing
  - This (the implementation that's in the PR) ☝🏽 actually works - though it is flaky